### PR TITLE
Support ListTrials by trial component name

### PR DESCRIPTION
**Pull request by l2yao**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-experiments/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-experiments/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-experiments/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-experiments/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/smexperiments/trial.py [hunk 1]
```diff
@@ -153,6 +156,7 @@ def list(
             api_types.TrialSummary.from_boto,
             "TrialSummaries",
             experiment_name=experiment_name,
+            trial_component_name=trial_component_name,
```
Comment (danabens) [inline src/smexperiments/trial.py]: does this depend on a new boto version? please verify min boto version specified in dependencies has this change

Comment (l2yao) [reply src/smexperiments/trial.py]: from https://github.com/boto/boto3/blob/2a4efd21345597189fbe3cccaa4b5593da504cb1/.changes/1.10.20.json looks like last update on sagemaker client is happening in 1.10.20 version

Comment (l2yao) [reply src/smexperiments/trial.py]: actually mini boto3 version is https://boto3.amazonaws.com/v1/documentation/api/1.12.8/reference/services/sagemaker.html#SageMaker.Client.list_trials

Comment (danabens) [reply src/smexperiments/trial.py]: looks like min boto version needs to be bumped then 
https://github.com/aws/sagemaker-experiments/blob/master/setup.py#L27

Comment (l2yao) [reply src/smexperiments/trial.py]: sure, updated

**Pull request merged**